tests/queries/0_stateless/02317_distinct_in_order_optimization_explain.reference

```diff
@@ -0,0 +1,19 @@
+-- disable optimize_distinct_in_order
+-- distinct all primary key columns -> no optimizations
+No optimizations
+-- enable optimize_distinct_in_order
+-- distinct with all primary key columns -> pre-distinct optimization only
+DistinctSortedChunkTransform
+-- distinct with primary key prefix -> pre-distinct optimization only
+DistinctSortedChunkTransform
+-- distinct with primary key prefix and order by on column in distinct -> pre-distinct and final distinct optimization
+DistinctSortedChunkTransform
+DistinctSortedChunkTransform
+-- distinct with primary key prefix and order by on column _not_ in distinct -> pre-distinct optimization only
+DistinctSortedChunkTransform
+-- distinct with non-primary key prefix -> no optimizations
+No optimizations
+-- distinct with non-primary key prefix and order by on column in distinct -> final distinct optimization only
+DistinctSortedChunkTransform
```

Comment:
I'd prefer to have a different name for final distinct, i.e. `DistinctSortedStreamTransform`. But the implementation is the same, so I'm not sure if it is worth the hassle to have a flag (i.e.`final_distinct`), and provide the different name if it's true. @KochetovNicolai any thoughts?